website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,100 @@
+
+![An example showcasing the Code Block paired with content in the nautral flow](/assets/components/code-block/code-block-dont-use-metadata.png)
+
+## CopyButton
+
+Use a CopyButton within the CodeBlock to make copying the snippet a single action. More details can be found in the [CopyButton guidelines](/components/copy/button).
+
+![An example of the Copy Button within the Code Block](/assets/components/code-block/code-block-copy-button.png)
+
+### Line selection
+
+If a user needs to copy only a portion of the CodeBlock, the relevant portion can be selected with a cursor and copied via the keyboard or mouse.
+
+![Selecting a single line in the Code Block](/assets/components/code-block/code-block-line-selection.png)
+
+## Line numbers
+
+Line numbers are displayed by default and can make longer blocks of code and snippets easier to parse. This is especially true in the case of logs and long configuration files that may have a higher degree of complexity.
+
+![Example of line numbers within the Code Block](/assets/components/code-block/code-block-line-numbers.png)
+
+!!! Info
+
+In the Figma component, the code examples have the appropriate number of lines by default, but must be manually hidden or shown to match the length of custom snippets.
+!!!
+
+## Line highlighting
+
+Use line highlighting to target and call attention to specific lines or multiple lines within a block of code.
+
+![Example of line highlighting in the Code Block](/assets/components/code-block/code-block-line-highlighting.png)
+
+!!! Info
+
+In the Ember component, lines can be highlighted by passing a single line number, multiple line numbers, or a range of lines. For more examples refer to the [How to use](/?tab=code#highlightlines) documentation. 
+
+In the Figma component, line highlighting (`highlightLines`) is a boolean property that displays a _representative_ example of what a highlighted line will look like visually and is not a property that can be controlled by the consumer. Instead, lines that are intended to be highlighted should be communicated in the engineering handoff or annotated in the design file.
+!!!
+
+## Language
+
+Language determines how syntax highlighting is applied and formatted within the block, but is handled a bit differently between the Ember and Figma components.
+
+The **Ember** component uses [Prism.js](https://prismjs.com/index.html) to handle syntax highlighting and supports close to [300 different languages and syntaxes](https://prismjs.com/index.html#supported-languages). The manner in which the syntax highlighting is applied is dependent on the value passed to the component.
```

Comment:
> supports close to [300 different languages and syntaxes

This is not true anymore, should be updated accordingly.

(unless you mean `Prism` in which case the paragraph is misleading: as a consumer, it doesn't matter how many are supported by Prims, what matters is how many are supported by the HDS component

Reply:
I guess I must have misunderstood something here related to the status autoloader, I'll adjust this to only include the languages outlined in the component.

Reply:
Something that I didn't get the chance to communicate to you, Jory: during the last engineering sync [we decided to bundle a pre-defined set of languages](https://docs.google.com/document/d/1twnk39hZ7p8oZYrld_FYsdiuUb6yn_weA46KnhF4nu4/edit#bookmark=id.vngglmevzfur) because of technical limitations with using the autoloader approach. We've kept [an item in the backlog to re-evaluate a more dynamic way of including languages](https://hashicorp.atlassian.net/browse/HDS-2660).

Reply:
Sounds good, thanks for the update! I may consider limiting the language examples in the Figma component as well to match this

Reply:
Yes, from what I've seen we mostly share the same languages, especially as the design audit informed many of these choices.

Reply:
Updated in [3f7c899](https://github.com/hashicorp/design-system/pull/1749/commits/3f7c8999e9932a662197bc26d785a8a3c477e091)